core/server/api_container/server/startosis_engine/kurtosis_instruction/run_sh/run_sh.go

```diff
@@ -0,0 +1,237 @@
+			WorkDirArgName:   true,
+			FilesAttr:        true,
+		},
+	}
+}
+
+type RunShCapabilities struct {
+	runtimeValueStore *runtime_value_store.RuntimeValueStore
+	serviceNetwork    service_network.ServiceNetwork
+
+	resultUuid string
+	name       string
+	run        string
+	image      string
+	workdir    string
+	files      map[string]string
+}
+
+func (builtin *RunShCapabilities) Interpret(arguments *builtin_argument.ArgumentValuesSet) (starlark.Value, *startosis_errors.InterpretationError) {
+	runCommand, err := builtin_argument.ExtractArgumentValue[starlark.String](arguments, RunArgName)
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", RunArgName)
+	}
+	builtin.run = runCommand.GoString()
+
+	if arguments.IsSet(ImageNameArgName) {
+		imageStarlark, err := builtin_argument.ExtractArgumentValue[starlark.String](arguments, ImageNameArgName)
+		if err != nil {
+			return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", ImageNameArgName)
+		}
+		builtin.image = imageStarlark.GoString()
+	}
+
+	if arguments.IsSet(WorkDirArgName) {
+		workDirStarlark, err := builtin_argument.ExtractArgumentValue[starlark.String](arguments, WorkDirArgName)
+		if err != nil {
+			return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", WorkDirArgName)
+		}
+		builtin.workdir = workDirStarlark.GoString()
+	}
+
+	if arguments.IsSet(FilesAttr) {
+		filesStarlark, err := builtin_argument.ExtractArgumentValue[*starlark.Dict](arguments, FilesAttr)
+		if err != nil {
+			return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", FilesAttr)
+		}
+		if filesStarlark.Len() > 0 {
+			filesArtifactMountDirpaths, interpretationErr := kurtosis_types.SafeCastToMapStringString(filesStarlark, FilesAttr)
+			if interpretationErr != nil {
+				return nil, interpretationErr
+			}
+			builtin.files = filesArtifactMountDirpaths
+		}
+	}
+
+	resultUuid, err := builtin.runtimeValueStore.CreateValue()
+	if err != nil {
+		return nil, startosis_errors.NewInterpretationError("An error occurred while generating UUID for future reference for %v instruction", RunShBuiltinName)
+	}
+	builtin.resultUuid = resultUuid
+	randomUuid := uuid.NewRandom()
+	builtin.name = fmt.Sprintf("task-%v", randomUuid.String())
+
+	dict := &starlark.Dict{}
+	err = dict.SetKey(starlark.String(runshCodeKey), starlark.String(fmt.Sprintf(magic_string_helper.RuntimeValueReplacementPlaceholderFormat, builtin.resultUuid, runshCodeKey)))
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "An error happened while creating exec return value, setting field '%v'", runshCodeKey)
+	}
+	err = dict.SetKey(starlark.String(runshOutputKey), starlark.String(fmt.Sprintf(magic_string_helper.RuntimeValueReplacementPlaceholderFormat, builtin.resultUuid, runshOutputKey)))
```

Comment:
I suggest to put this: `fmt.Sprintf(magic_string_helper.RuntimeValueReplacementPlaceholderFormat, builtin.resultUuid, runshOutputKey)))` in a var for improving legibility